ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -572,6 +586,14 @@ monitorBootstrapPeersFlag PeerSelectionActions { readUseBootstrapPeers }
              , ledgerStateJudgement  = YoungEnough
```

Comment:
This is what is written in the function's haddock:

> -- | Monitor 'UseBootstrapPeers' flag.
--
-- The user might reconfigure the node at any point to change the value of
-- UseBootstrapPeers. Essentially the user can enable or disable bootstrap
-- peers at any time. Since 'monitorLedgerStateJudgement' will act on the
-- ledger state judgement value changing, this monitoring action should only
-- be responsible for either disabling bootstrap peers (in case the user
-- disables this flag) or enabling the 'monitorLedgerStateJudgement' action to
-- work correctly in the case the node finds itself in bootstrap mode. In
-- order to achieve this behavior, when bootstrap peers are disabled we should
-- update the ledger state judgement value to 'YoungEnough'; and
-- 'hasOnlyBootstrapPeers' to 'False'.
--
-- Here's a brief explanation why this works. There's 4 scenarios to consider:
--
-- 1. The node is in 'YoungEnough' state and the user
--  1.1. Enables bootstrap peers: In this case since the node is caught up
--  nothing should happen, so setting the LSJ to YoungEnough state is
--  idempotent.
--  1.2. Disables bootstrap peers: In this case, since the node is caught up,
--  its functioning can't really be distinguished from that of a node that has
--  bootstrap peers disabled. So changing the LSJ and 'hasOnlyBootstrapPeers'
--  flag is idempotent.
-- 2. The node is in 'TooOld' state and the user
--  2.1. Enables bootstrap peers: If the node is behind, enabling bootstrap
--  peers will enable 'monitorLedgerStateJudgement'. So if we set the LSJ to
--  be in 'YoungEnough' state it is going to make sure 'monitorLedgerStateJudgement'
--  observes the 'TooOld' state, triggering the right measures to be taken.
--  2.2. Disables bootstrap peers: If this is the case, we want to let the
--  peer connect to non-trusted peers, so just updating the boostrap peers
--  flag will enable the previously disabled monitoring actions.
--

Reply:
Thanks Armando.

> -- 2.1. Enables bootstrap peers: If the node is behind, enabling bootstrap
> -- peers will enable 'monitorLedgerStateJudgement'. So if we set the LSJ to
> -- be in 'YoungEnough' state it is going to make sure 'monitorLedgerStateJudgement'
> -- observes the 'TooOld' state, triggering the right measures to be taken.

Hmm, it's a bit strange that we set it to `YoungEnough` and then `TooOld` is observed, why is it so?

We're breaking principle that there should be one source of truth (the value of ledgerStateJudgement can come  either from here or from consensus).

/it would be much simpler if bootstrap peers were a static property, maybe we complicated it for ourselves/